src/components/form.js

```diff
@@ -1,44 +1,48 @@
 import React, { useState } from 'react';
-import PropTypes from 'prop-types';
+import { v4 as uuidv4 } from 'uuid';
+import { useDispatch } from 'react-redux';
+import { createBook } from '../redux/api/api';
 
-const Form = ({ addBook }) => {
-  const [book, setBook] = useState({});
+const Form = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [category, setCategory] = useState('Action');
 
-  const onChange = (e) => {
-    setBook({ ...book, [e.target.name]: [e.target.value] });
-  };
-
-  const handleSubmit = (e) => {
-    addBook(book);
-    book.title = '';
-    book.author = '';
+  const submitBookToStore = (e) => {
     e.preventDefault();
+    const book = {
+      item_id: uuidv4(),
+      title,
+      category,
+    };
+    createBook(book, dispatch);
```

Comment:
- [OPTIONAL] Although, this implementation works fine, I think this syntax is not the best way to handle asynchronous requests with redux. Usually, you would have this pattern
1- Dispatch the asynchronous action (For example here it would be `dispatch(createBook)`)
2- The asynchronous action after executing and getting the response will dispatch one or many synchronous actions according to the promise status `(fulfilled or rejected)`

As an example we can have something like:
```
export const createBook = (newBook) => async (dispatch) => {
  axios
    .post(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/KP5PI7hEkMIt6Rn76UzC/books',
      newBook,
    )
    .then((res) => {
      if (res.status === 201) {
        dispatch(addBook(newBook));
      }
    })
    .catch((error) => error);
};
```
- [OPTIONAL] Refactor all your actions in `src/redux/api/api.js` to follow the example
- [OPTIONAL] Refactor the asynchronous actions calls in your components
`loadBooks(dispatch) --> dispatch(loadBooks())`, 
`deleteBook(btnId, dispatch) --> dispatch(deleteBook(bookId))`, 
` createBook(book, dispatch) --> dispatch(createBook(newBook))`